torch/testing/_internal/composite_compliance.py

```diff
@@ -410,3 +442,58 @@ def check_backward_formula(op, args, kwargs):
                     f"- wrapped_kwargs: {which_kwargs_are_wrapped}\n"
                     f"- wrapped_grads: {which_grad_is_batched}\n"
                 )
+
+# Checks if the forward AD formula is composite compliant by testing
+# all possible permutations of {primals, tangents} being
+# CompositeCompliantTensor or regular Tensors.
+def check_forward_formula(op, args, kwargs):
+    assert op.supports_forward_ad
+
+    # Permutations of arg and kwargs in CCT.
+    for choice in generate_subclass_choices_args_kwargs(args, kwargs):
+        new_args, new_kwargs, which_args_are_wrapped, which_kwargs_are_wrapped = choice
+
+        def maybe_tangent(t):
+            # Generate `tangent` tensor
+            # if given object is a Tensor
+            if isinstance(t, torch.Tensor):
+                return torch.empty_like(t)
+            elif is_tensorlist(t):
+                return list(torch.empty_like(e) for e in t)
```

Comment:
The tangent should really not be empty_like, empty_like creates a Tensor with uninitialized memory. Let's change this to randn_like?